packages/flutter_test/test/matchers_test.dart

```diff
@@ -721,6 +717,427 @@ void main() {
+         /* Actions */
+         hasTapAction: true,
+         hasLongPressAction: true,
+         hasScrollLeftAction: true,
+         hasScrollRightAction: true,
+         hasScrollUpAction: true,
+         hasScrollDownAction: true,
+         hasIncreaseAction: true,
+         hasDecreaseAction: true,
+         hasShowOnScreenAction: true,
+         hasMoveCursorForwardByCharacterAction: true,
+         hasMoveCursorBackwardByCharacterAction: true,
+         hasMoveCursorForwardByWordAction: true,
+         hasMoveCursorBackwardByWordAction: true,
+         hasSetTextAction: true,
+         hasSetSelectionAction: true,
+         hasCopyAction: true,
+         hasCutAction: true,
+         hasPasteAction: true,
+         hasDidGainAccessibilityFocusAction: true,
+         hasDidLoseAccessibilityFocusAction: true,
+         hasDismissAction: true,
+         customActions: <CustomSemanticsAction>[action],
+      ));
+    });
+
+    testWidgets('can match all flags and actions disabled', (WidgetTester tester) async {
+      final SemanticsData data = SemanticsData(
+        flags: 0,
+        actions: 0,
+        attributedLabel: AttributedString('a'),
+        attributedIncreasedValue: AttributedString('b'),
+        attributedValue: AttributedString('c'),
+        attributedDecreasedValue: AttributedString('d'),
+        attributedHint: AttributedString('e'),
+        tooltip: 'f',
+        textDirection: TextDirection.ltr,
+        rect: const Rect.fromLTRB(0.0, 0.0, 10.0, 10.0),
+        elevation: 3.0,
+        thickness: 4.0,
+        textSelection: null,
+        scrollIndex: null,
+        scrollChildCount: null,
+        scrollPosition: null,
+        scrollExtentMax: null,
+        scrollExtentMin: null,
+        platformViewId: 105,
+        currentValueLength: 10,
+        maxValueLength: 15,
+      );
+      final _FakeSemanticsNode node = _FakeSemanticsNode(data);
+
+      expect(node, containsSemantics(
+         rect: const Rect.fromLTRB(0.0, 0.0, 10.0, 10.0),
+         size: const Size(10.0, 10.0),
+         elevation: 3.0,
+         thickness: 4.0,
+         platformViewId: 105,
+         currentValueLength: 10,
+         maxValueLength: 15,
+         /* Flags */
+         hasCheckedState: false,
+         isChecked: false,
+         isSelected: false,
+         isButton: false,
+         isSlider: false,
+         isKeyboardKey: false,
+         isLink: false,
+         isTextField: false,
+         isReadOnly: false,
+         hasEnabledState: false,
+         isFocused: false,
+         isFocusable: false,
+         isEnabled: false,
+         isInMutuallyExclusiveGroup: false,
+         isHeader: false,
+         isObscured: false,
+         isMultiline: false,
+         namesRoute: false,
+         scopesRoute: false,
+         isHidden: false,
+         isImage: false,
+         isLiveRegion: false,
+         hasToggledState: false,
+         isToggled: false,
+         hasImplicitScrolling: false,
+         /* Actions */
```

Comment:
The (helpful) repeated commenting of actions/flags makes me think we should eventually group these into different objects to cut back on the number of parameters to this function and make their grouping clearer.

Reply:
Yeah... I can agree with that. I was thinking something like this:

```
containsSemantics(
  expectedActions: [<insert SemanticAction values here>],
  unexpectedActions: [<insert SemanticAction values here>],
  expectedFlags: [<insert SemanticFlag values here>],
  unexpectedFlags: [<insert SemanticFlag values here>],
);
```

But I leaned towards keeping the API the same between the two. That said, I was debating setting up `containsSemantics` as a full on replacement at some point by eventually deprecating `matchesSemantics`, so this would be the time to modify the API if we're going to do it.

@goderbauer thoughts?

Reply:
Yeah, I don't think it should be done in this PR, but it would be a nice QoL change.

Reply:
The issue being that if we _don't_ do it in this PR, then we're introducing a breaking change later on. If we use a different syntax for `containsSemantics` now, it'll be different from `matchesSemantics` if we don't deprecate that value later on, but it doesn't require a breaking change anywhere (minus the possible deprecation at a later date).

Reply:
Here's my guess on the reasoning for the individual booleans. Looking at the Semantics widget (the usual entry point for people doing any semantics work) these values aren't exposed as the SemanticsAction or SemanticsFlag values, they're just bools or otherwise implicitly defined through other parameters. As such, as troublesome as it is in _this_ part of the code, it's much more familiar for the people _using_ the actual matcher as opposed to testing it.

With that in mind, I'm leaning back towards keeping the booleans, even though they're a bit annoying to spell out when you're doing extreme cases like these tests are.

Reply:
I am open to improving the API, but from the perspective of a developer coming from working with the semantics widget, the booleans do feel more natural...